transcom/mymove
# B 19813 - Customer time validation - HHG, NTS, NTSR

## [B-19813](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A946350&RoomContext=TeamRoom%3A855664&concept=TeamRoom)

## Summary

**INTEGRATION**

This PR adds date validation for HHG, NTS and NTSR for preferred pickup and/or delivery date selection for the shipment onboarding process. An inline alert message will appear upon date selection from date picker. An alert message will appear on page load and on date selection. Message will be customized to respective date label. Alert will notify user if date selected is on weekend and/or holiday(federal holidays). For now implementation will assume date context is in the United States. For example, July 4th is a holiday and Sat/Sunday are considered weekend days.   

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Alerts will only display if date is on a weekend and/or holday.

1. Login as customer
2. Add shipment types: HHG, NTS, NTSR
3. For each shipment verify alert messages are displayed when holidays and weekend dates are selected for preferred pickup and delivery dates during onboarding.
4. Verify alerts are displayed on shipment edit.
5. Verify alert messages are for pickup or delivery.

![Screenshot 2024-08-06 at 1 19 00 PM](https://github.com/user-attachments/assets/f19243b1-6767-4248-aff5-da3dcb955684)
![Screenshot 2024-08-06 at 12 16 55 PM](https://github.com/user-attachments/assets/d3865d4c-9b2c-4387-8062-3799ef40fa74)
![Screenshot 2024-08-06 at 12 16 45 PM](https://github.com/user-attachments/assets/2833e911-d1ca-4a34-b3ac-ac44146664d5)


### Frontend


### Backend
